heyoka/test.py

```diff
@@ -620,6 +620,26 @@ def cb1(ta, mr):
             self.assertEqual(counter_nt, 3)
             self.assertEqual(counter_t, 2)
 
+class expression_eval_test_case(_ut.TestCase):
+    def runTest(self):
+            from . import sin, make_vars, with_real128, eval
+            import numpy as np
+
+            x, = make_vars("x")
+
+            fp_types = [("double", float), ("long double", np.longdouble)]
+
+            if with_real128:
+                from mpmath import mpf
+                fp_types.append(("real128", mpf))
+
+            for desc, fp_t in fp_types:
+                target = fp_t("0.123456789012345678901234567890")
+                a = eval(x, {"x": target}, fp_type=desc)
+                self.assertEqual(a, target)
+                a = eval(x**3.1, {"x": target}, fp_type=desc)
+                self.assertAlmostEqual(a, target**3.1, places=14)
```

Comment:
The ``places=`` argument in this test should depend on the floating-point type in use. I am trying to understand how to fetch the properties of the long double type.

Reply:
Ok for ``double`` and ``long double`` you can use ``numpy.floatinfo``:

```ipython
In [28]: x = np.longdouble(0)

In [29]: np.finfo(x)
Out[29]: finfo(resolution=1e-18, min=-1.189731495357231765e+4932, max=1.189731495357231765e+4932, dtype=float128)

In [30]: np.finfo(1.2)
Out[30]: finfo(resolution=1e-15, min=-1.7976931348623157e+308, max=1.7976931348623157e+308, dtype=float64)
```

For quad precision you can hard-code something like 1e-34 as the epsilon.

Reply:
something like that, but for ``longdouble`` you should still fetch the epsilon in the test without hard coding to 17, because on Windows ``longdouble == double``.